### PR TITLE
add new package : filebench

### PR DESCRIPTION
**Pull request by darmac**
@adamjstewart 
Add a filebench package. however, it's a little complex in autoreconf() function, I'm not sure this style is OK or not.

### CODE REVIEW

File: var/spack/repos/builtin/packages/filebench/package.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    and database-server workloads). Filebench is equally good for micro
+    and macro-benchmarking, quick to setup, and relatively easy to use.
+    """
+
+    homepage = "https://github.com/filebench/filebench"
+    url      = "https://github.com/filebench/filebench/archive/1.5-alpha3.tar.gz"
+
+    version('1.5-alpha3', sha256='2dedfc46458f5bb13e5f5a4a9f4db6263152f6186690d57653c96138610db35a')
+    version('1.5-alpha2', sha256='02aeae7bd7e6c3e1d32862a4d1c5e14e59a1ffbb8fadbf09dc84ebe82bc6029e')
+    version('1.5-alpha1', sha256='8266c25274a20025c325050ed4050d2783d239f77558fc332381d05de9917dfc')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/filebench/package.py]: You don't have to add these alpha versions if you don't need them

File: var/spack/repos/builtin/packages/filebench/package.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    depends_on('m4',       type='build')
+    depends_on('flex',     type='build')
+    depends_on('bison',    type='build')
+
+    def autoreconf(self, spec, prefix):
+        sh = which('sh')
+        sh('libtoolize')
+        sh('aclocal')
+        sh('autoheader')
+        sh('autoheader')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/filebench/package.py]: Why does this need to be run twice?

File: var/spack/repos/builtin/packages/filebench/package.py (hunk 1 above)
Comment (darmac) [reply var/spack/repos/builtin/packages/filebench/package.py]: fix this in https://github.com/spack/spack/pull/14158/commits/deca899816b4916565daf8930e0a62b2bd3a31b6

File: var/spack/repos/builtin/packages/filebench/package.py (hunk 2 above)
Comment (darmac) [reply var/spack/repos/builtin/packages/filebench/package.py]: https://github.com/spack/spack/pull/14158/commits/deca899816b4916565daf8930e0a62b2bd3a31b6

File: var/spack/repos/builtin/packages/filebench/package.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    flexible and allows to specify application's I/O behavior using its
+    extensive Workload Model Language (WML). Users can either describe
+    desired workloads from scratch or use(with or without modifications)
+    workload personalities shipped with Filebench(e.g., mail-, web-, file-,
+    and database-server workloads). Filebench is equally good for micro
+    and macro-benchmarking, quick to setup, and relatively easy to use.
+    """
+
+    homepage = "https://github.com/filebench/filebench"
+    url      = "https://github.com/filebench/filebench/archive/1.5-alpha3.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/filebench/package.py]:
```suggestion
    url      = "https://github.com/filebench/filebench/archive/1.4.9.1.tar.gz"
```

File: var/spack/repos/builtin/packages/filebench/package.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    depends_on('flex',     type='build')
+    depends_on('bison',    type='build')
+
+    def autoreconf(self, spec, prefix):
+        sh = which('sh')
+        sh('libtoolize')
+        sh('aclocal')
+        sh('autoheader')
+        sh_automake = Executable('automake --add-missing')
+        sh_automake()
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/filebench/package.py]:
```python
sh('automake', '--add-missing')
```

Comment (darmac) [reply var/spack/repos/builtin/packages/filebench/package.py]: fix it in https://github.com/spack/spack/pull/14158/commits/1c6d4df10173ef85695a8f109f5894cd26343b9d

File: var/spack/repos/builtin/packages/filebench/package.py (hunk 3 above)
Comment (darmac) [reply var/spack/repos/builtin/packages/filebench/package.py]: fix it in https://github.com/spack/spack/pull/14158/commits/10d76007d82d38cd02bb33b63eedb9032074eacf

**Comment by adamjstewart** [comment]
Closing and reopening to try to fix Travis checks

**Pull request merged**